android/app/src/main/java/org/jitsi/meet/MainActivity.java

```diff
@@ -1,46 +1,98 @@
+ */
+public class MainActivity extends AppCompatActivity {
+    private JKConferenceView conferenceView;
+    public static final int OVERLAY_PERMISSION_REQ_CODE = 4242;
+
     @Override
-    protected ReactActivityDelegate createReactActivityDelegate() {
-        return new ReactActivityDelegate(this, getMainComponentName()) {
-            /**
-             * {@inheritDoc}
-             *
-             * Overrides {@link ReactActivityDelegate#createRootView()} to
-             * customize the {@link ReactRootView} with a background color that
-             * is in accord with the JavaScript and iOS parts of the application
-             * and causes less perceived visual flicker than the default
-             * background color.
-             */
-            @Override
-            protected ReactRootView createRootView() {
-                ReactRootView rootView = super.createRootView();
-
-                rootView.setBackgroundColor(0xFF111111);
-                return rootView;
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (requestCode == OVERLAY_PERMISSION_REQ_CODE) {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                if (!Settings.canDrawOverlays(this)) {
+                    // Permission not granted...
+                    return;
+                }
             }
-        };
+
+            setContentView(conferenceView);
+        }
+    }
+
+
+    @Override
+    public void onBackPressed() {
+        if (!JKConferenceView.onBackPressed()) {
+            // Invoke the default handler if it wasn't handled by React.
+            super.onBackPressed();
+        }
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        conferenceView = new JKConferenceView(this);
+        conferenceView.loadURL(null);
+
+        /**
+         * In debug mode React needs permission to write over other apps in order to display the
+         * warning and error overlays.
+         */
+        if (BuildConfig.DEBUG && Build.VERSION.SDK_INT >= Build.VERSION_CODES.M &&
+            !Settings.canDrawOverlays(this)) {
+
+            Intent intent
+                = new Intent(Settings.ACTION_MANAGE_OVERLAY_PERMISSION,
+                    Uri.parse("package:" + getPackageName()));
+            startActivityForResult(intent, OVERLAY_PERMISSION_REQ_CODE);
+            return;
+        }
+
+        setContentView(conferenceView);
     }
 
-    /**
-     * Returns the name of the main component registered from JavaScript.
-     * This is used to schedule rendering of the component.
-     */
     @Override
-    protected String getMainComponentName() {
-        return "App";
+    public void onNewIntent(Intent intent) {
+        JKConferenceView.onNewIntent(intent);
     }
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        JKConferenceView.onHostDestroy(this);
```

Comment:
The docs got scrapped when I added the view, I'll brind those back when I merge the approaches. In a nutshell, this is not about deep linking, but about activity management. Those activity lifecycle methods need to be tied to React Native.

I'm happy to move those elsewhere, but they need to call stuff on the React native Instance Manager, of which we hold a single instance. That's why I put them in the view as static methods.

I'll make this clear in the docs.